### PR TITLE
Merge upstream

### PR DESCRIPTION
**Pull request by EmelyanenkoK**

### CODE REVIEW

**Comment by MXzander** [comment]
8a6277a8da77a3aa29f14706c720ce3d18497edc

File: bin/generate-declarations [hunk 1]
```diff
@@ -0,0 +1,38 @@
+#!/usr/bin/env bash
+
+set -e
```
Comment (hossen71) [inline bin/generate-declarations:3]: UQA5NWl5eH6KvxwnMBCj7vDbb-Xebok-pWGfFuHhq1zMxmv5

File: src/utils/Utils.js [hunk 2]
```diff
@@ -0,0 +1,325 @@
...
+
+module.exports = {
+    BN,
+    nacl,
+    sha256,
+    fromNano,
+    toNano,
+    bytesToHex,
+    hexToBytes,
+    stringToBytes,
```
Comment (hossen71) [inline src/utils/Utils.js:315]: FF00F4A413F4BCF2C80B} 
x{2_} 
  x{4} 
   x{D001D0D3032171B0925F04E022D749C120925F04E002D31F218210706C7567BD22821064737472BDB0925F05E003FA403020FA4401C8CA07CBFFC9D0ED44D0810140D721F404305C810108F40A6FA131B3925F07E005D33FC8258210706C7567BA923830E30D03821064737472BA925F06E30D} 
    x{01FA00F40430F8276F2230500AA121BEF2E0508210706C7567831EB17080185004CB0526CF1658FA0219F400CB6917CB1F5260CB3F20C98040FB0006} 
    x{5004810108F45930ED44D0810140D720C801CF16F400C9ED540172B08E23821064737472831EB17080185005CB055003CF1623FA0213CB6ACB1FCB3FC98040FB00925F03E2} 
   x{2_} 
    x{2_} 
     x{5} 
      x{B29DFB513420405035C87D010C00B23281F2FFF274006040423D029BE84C6_} 
      x{2_} 
       x{ADCE76A26840206B90EB85FFC_} 
       x{AF1DF6A26840106B90EB858FC_} 
     x{B8C97ED44D0D70B1F} 
    x{BD242B6F6A2684080A06B90FA0218470D4080847A4937D29910CE6903E9FF9837812801B7810148987159F3184_} 
  x{F28308D71820D31FD31FD31F02F823BBF264ED44D0D31FD31FD3FFF404D15143BAF2A15151BAF2A205F901541064F910F2A3F80024A4C8CB1F5240CB1F5230CBFF5210F400C9ED54F80F01D30721C0009F6C519320D74A96D307D402FB00E830E021C001E30021C002E30001C0039130E30D03A4C8CB1F12CB1FCBFF} 
   x{D207FA00D4D422F90005C8CA0715CBFFC9D077748018C8CB05CB0222CF165005FA0214CB6B12CCCCC973FB00C84014810108F451F2A702} 
   x{810108D718FA00D33FC8542047810108F451F2A782106E6F746570748018C8CB05CB025006CF165004FA0214CB6A12CB1FCB3FC973FB0002} 
   x{810108D718FA00D33F305224810108F459F2A782106473747270748018C8CB05CB025005CF165003FA0213CB6ACB1F12CB3FC973FB00} 
   x{F400C9ED54}
داده‌ها
سلول
هگز
پایه64
x{000000A229A9A3173666ED1E59CAE914CB8B381FB12ABA80C205AA071771D024A0B61B7210107A21

**Review by hossen71 — COMMENTED** [review]
8a6277a8da77a3aa29f14706c720ce3d18497edc

**Pull request merged**